lib/note-content-editor.jsx

```diff
@@ -171,6 +220,29 @@ export default class NoteContentEditor extends React.Component {
 		)
 	}
 
+	handleReturn = () => {
+		// matches lines that start with `- ` or `* `
+		// preceded by 0 or more tab characters
+		const listItemRe = /^\t*[-*]\s/;
```

Comment:
1. in the mac app at least yes it it is a valid list even if if it has no indentation
2. I made it work only with tab characters for simplicity but it can be changed